epython/portfolio_management/cash/interest_rate.py

```diff
@@ -0,0 +1,125 @@
+import kydb
+from portfolio_management.kydb_config import OBJDB_CONFIG
+from portfolio_management.utils.sql import SQLMixin
+from portfolio_management.common.base_item import Factory
+import hashlib
+import pandas as pd
+
+QUERY_RESULT_CACHE = '/cache/'
+
+class InterestRateHistory(SQLMixin, kydb.DbObj):
+
+
+
+    @kydb.stored
+    def path(self) -> str:
+        return ''
+
+    @kydb.stored
+    def history(self) -> list:
+        return []
+
+SQL_TEMPLATE = '''
+WITH t AS
+    (SELECT date_parse(datetime,
+        '%m/%d/%Y') AS dt
+    FROM interest_rate
+    WHERE pair = '{pair}'
+            AND year = '{dt:%Y}'
+SELECT *
+FROM t
+WHERE dt = date = '{dt:%Y-%m-%d}'        
+'''
+
+
+class InterestRate(SQLMixin, kydb.DbObj):
+    """
+    Interest rate
+    Ideally we could updated tis daily from some source
+    it should be independent for each currency
+    """
+
+    @staticmethod
+    @kydb.stored
+    def day_count():
+        return 365
+
+    @kydb.stored
+    def res_s3_path(self):
+        return None
+
+    def sql(self, dt, ccy):
+        dt = self._ensure_datetime(dt)
+
+        kwargs = {'dt': dt, 'ccy': ccy}
+
+        for k, v in self.sql_kwargs().items():
+            kwargs[k] = v
+
+        return self.sql_template().format(**kwargs)
```

Comment:
Shouldn't expose SQL logic in the interest rate class

Reply:
Agree, it is a left over form when I copied the initial class, I will get rid of it. As mentioned in slack, this is the class I spent less time on, I was trying to get everything else to a semi working stage. Will have a look.

Reply:
Cool. I'll spend more time tomorrow going through the PR.

Reply:
AT the moment interest rate is hardcoded in account eod process (interest_rate)

Reply:
Forgot to add, if you look at test_portfolio, that loads all object types to a portfolio